pkgs/development/compilers/llvm/3.8/llvm.nix

```diff
@@ -41,25 +40,29 @@ in stdenv.mkDerivation rec {
     ln -sv $PWD/lib $out
   '';
 
-  cmakeFlags = with stdenv; [
+  cmakeFlags = with stdenv;
+  [
     "-DCMAKE_BUILD_TYPE=${if debugVersion then "Debug" else "Release"}"
     "-DLLVM_INSTALL_UTILS=ON"  # Needed by rustc
     "-DLLVM_BUILD_TESTS=ON"
     "-DLLVM_ENABLE_FFI=ON"
     "-DLLVM_ENABLE_RTTI=ON"
-  ] ++ stdenv.lib.optional enableSharedLibraries [
-    "-DLLVM_LINK_LLVM_DYLIB=ON"
-  ] ++ stdenv.lib.optional (!isDarwin)
+  ] ++ stdenv.lib.optional enableSharedLibraries
+    "-DBUILD_SHARED_LIBS=ON"
```

Comment:
I believe this causes many .so files to be produced instead of a single libLLVM.so. This causes poor performance for applications which want to link against LLVM.